components/raftstore/src/store/entry_storage.rs

```diff
@@ -1230,6 +1342,7 @@ impl<EK: KvEngine, ER: RaftEngine> EntryStorage<EK, ER> {
             // should be temporarily skipped. Otherwise, the warmup task may fail.
             return;
         }
+        self.term_cache.compact_to(idx);
```

Comment:
Should we reset `term_cache` during applying snapshot?

Reply:
Both accept, because the next `term_cache.append()` keeps safety of the consequential accessing of the `term()`.

Reply:
My concern is that `term_cache` may return a term for a truncated log which is inconsistent to the actual result `None` of raft engine, and this may lead to unexpected behavior.

Reply:
It should not. The `term_cache` lifecycle aligns with its corresponding term and will be compacted by the Leader during `RaftLogGc`.

For additional safety, I’ve moved the `term_cache.append` call into `handle_raft_committed_entries`. This ensures all cached entries are valid, as they are now only appended after being committed to a quorum of peers.